internal/search/query/predicate.go

```diff
@@ -318,6 +344,78 @@ func (p *RepoHasKeyPredicate) Unmarshal(params string, negated bool) (err error)
 func (p *RepoHasKeyPredicate) Field() string { return FieldRepo }
 func (p *RepoHasKeyPredicate) Name() string  { return "has.key" }
 
+// RepoContainsPredicate represents the `repo:contains(file:a content:b)` predicate.
+// DEPRECATED: this syntax is deprecated in favor of `repo.contains.file`.
+type RepoContainsPredicate struct {
+	File    string
+	Content string
+	Negated bool
+}
+
+func (f *RepoContainsPredicate) Unmarshal(params string, negated bool) error {
+	nodes, err := Parse(params, SearchTypeRegex)
+	if err != nil {
+		return err
+	}
+	for _, node := range nodes {
+		if err := f.parseNode(node); err != nil {
+			return err
+		}
+	}
+
+	if f.File == "" && f.Content == "" {
+		return errors.New("one of file or content must be set")
+	}
+	f.Negated = negated
+	return nil
+}
+
+func (f *RepoContainsPredicate) parseNode(n Node) error {
+	switch v := n.(type) {
+	case Parameter:
+		if v.Negated {
+			return errors.New("predicates do not currently support negated values")
+		}
+		switch strings.ToLower(v.Field) {
+		case "file":
+			if f.File != "" {
+				return errors.New("cannot specify file multiple times")
+			}
+			if _, err := regexp.Compile(v.Value); err != nil {
+				return errors.Errorf("`contains` predicate has invalid `file` argument: %w", err)
```

Comment:
It's not clear to the user that `contains` is scoped to the `repo:` filter here

```suggestion
				return errors.Errorf("the `repo:contains()` predicate has invalid `file` argument: %w", err)
```